### ISSUE TITLE
Auto-format files

### ISSUE THREAD

**Issue by Robinlovelace** [nptscot/nptscot.github.io#291]

**Comment by Robinlovelace** [comment]
I came across this auto-formatter and thought I would give it a go: https://github.com/biomejs/biome

That could help ensure code consistency and encourage contributions.

Just an idea, I know good style can be subjective but would be good to have a defined style and stick to it I guess.

**Comment by mvl22** [comment]
This tooling stuff needs far more configuration than just chucking it at a codebase. For instance it's gone and formatted a pile of library files (which are currently embedded in the code because we aren't using package management for deployment simplicity, though I don't like that). Some of the CSS looks pretty ugly and using a rare layout.

Yeah, it's subjective stuff; personally I would give this a miss for now as really this makes sense when there's a proper build chain, and ultimately using a definition in the editor is a better way to enforce this.

**Comment by Robinlovelace** [comment]
Maybe we could just auto-format the .js code as the example suggests: 

```
npx @biomejs/biome lint --write ./src
```

**Comment by Robinlovelace** [comment]
It's got a nice test website where you can interactively tweak the format settings to get the subjective + consistent style you like. At the moment the style is not fully consistent as far as I can tell. It would be nice if it were internally consistent, right?

![image](https://github.com/user-attachments/assets/781817e5-853b-4833-a680-de7550f87b3a)

Source: https://biomejs.dev/playground/

**Comment by Robinlovelace** [comment]
> a definition in the editor is a better way to enforce this

configs for linters like biome define the style for editors I think, e.g. with

```json
{
  "$schema": "https://biomejs.dev/schemas/1.9.4/schema.json",
  "vcs": {
    "enabled": false,
    "clientKind": "git",
    "useIgnoreFile": false
  },
  "files": { "ignoreUnknown": false, "ignore": [] },
  "formatter": { "enabled": true, "indentStyle": "tab" },
  "organizeImports": { "enabled": true },
  "linter": {
    "enabled": true,
    "rules": { "recommended": true }
  },
  "javascript": { "formatter": { "quoteStyle": "double" } }
}
```

Source: https://biomejs.dev/guides/getting-started/

Many [editors](https://biomejs.dev/guides/integrate-in-editor/) are supported, and there's a VS Code [extension](https://biomejs.dev/reference/vscode/) for biome, for example.

**Comment by mvl22** [comment]
Yes, that's what I mean that this is best handled at editor level, i.e. you add a linter definition and the editor handles this stuff. I'm not in favour of adding build processes to a site unnecessarily.

I did a lot of tidying up of the original indentation while doing the refactoring. There may well be inconsistencies still, yes, because developing a linting spec hasn't been a high priority.

I suggest you dogfood this on the R code first and see whether you find this stuff helpful or a pain...

**Comment by Robinlovelace** [comment]
> I suggest you dogfood this on the R code first and see whether you find this stuff helpful or a pain...

Super helpful on R side, we use the `styler.equals` auto styler.

**Comment by mvl22** [comment]
OK, I'll find a suitable VSCode linter spec and put that in when I get a chance, comparing biome and the one we use quite effectively on the whitelabel codebase.

**Comment by Robinlovelace** [comment]
Closing for now, we can revisit later.

**Comment by mvl22** [comment]
I'd like to keep this open, would be sensible to fix.

**Comment by Robinlovelace** [comment]
👍
